Detect RV300 and VC4 hardware and disable slow half float vertex on them

illwieckz opened the pull request:
Import from:

- https://github.com/DaemonEngine/Daemon/pull/1179

illwieckz commented:
I also added VC4 detection, it's twice faster with float vertex on it.

Before (25fps):

![unvanquished_2024-10-11_183704_000](https://github.com/user-attachments/assets/7d1b0031-338a-4cea-a617-fc7f713de52a)

After (55fps):

![unvanquished_2024-10-11_184004_000](https://github.com/user-attachments/assets/a196f051-3f99-469b-98f6-0bfbd6014278)

VReaperV commented on src/engine/sys/sdl_glimp.cpp:
```diff
@@ -2210,10 +2214,65 @@ static void GLimp_InitExtensions()
+		{
+			bool foundRv300 = false;
+
+			std::string cardName = "unknown ATI RV3xx";
```
The initial value is unused.

illwieckz replied:
It is used if I forgot to list a card name and the `glConfig.hardwareType == glHardwareType_t::GLHW_R300 && glConfig2.maxAluInstructions == 64` test matches.

slipher replied:
It isn't. It's assigned the first time the for loop is entered.

illwieckz replied:
Oh I see. Right.

slipher commented on src/engine/sys/sdl_glimp.cpp:
```diff
@@ -2210,10 +2214,65 @@ static void GLimp_InitExtensions()
+
+		if ( !halfFloatVertexEnabled )
+		{
+			logger.Warn( "Missing half-float vertex, using float vertex instead." );
```
How about Notice for this one. We shouldn't use Warn so much unless there is a serious problem.

illwieckz replied:
I like to have a bolder log for such things, especially when it is reporting about making decisions, it makes it easier to catch them in logs. Though I agree Warn is maybe too much.

illwieckz commented:
I addressed the various comments.

slipher commented:
> I also added VC4 detection, it's twice faster with float vertex on it.

Forgot to push something? I don't see any new stuff.

illwieckz commented:
> > I also added VC4 detection, it's twice faster with float vertex on it.
> 
> Forgot to push something? I don't see any new stuff.

Indeed, now fixed.

slipher commented:
LGTM
